### PR TITLE
chore: pull transifex translations automatically

### PR DESCRIPTION
**Pull request by SgtPooki**
This github action, .github/workflows/tx-pull.yml, will

  1. automatically pull translations
  2. automatically create a commit to a `tx-pull-<github_sha>` branch
  3. automatically open a PR from that branch to the main branch

### CODE REVIEW

File: .github/workflows/tx-pull.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        path: public
+        key: ${{ github.sha }}
+    - name: Install Transifex client
+      run: |
+        curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
+    - name: Pull translations from Transifex
+      run: |
+        tx -t ${{ secrets.TX_TOKEN }} pull -a
+
+  pull-request:
```
Comment (lidel) [inline .github/workflows/tx-pull.yml:23]: Do we need two jobs here? iiuc everything could be done in one, removing need for passing things in cache

**Review by lidel — CHANGES_REQUESTED** [review]
lgtm, but running it on every push is too much, suggestions inline

File: .github/workflows/tx-pull.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+on:
+  push:
+    branches:
+    - main
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply .github/workflows/tx-pull.yml:4]: The only problem with this, is that I don't think the github action I'm using for creating PRs supports updating PRs. Do you know of a github action that would accomplish that?

[the remark this answers is not in the document]
Comment (lidel) [reply .github/workflows/tx-pull.yml]: I think `repo-sync/pull-request` does force-push if branch already exists?

We use it [here](https://github.com/ipfs/ipfs-docs/blob/main/.github/workflows/update-on-new-ipfs-tag.yml#L23), and it seems to force-push [here](https://github.com/ipfs/ipfs-docs/pull/1174), so maybe it already works but is not documented?

I agree there is no point in spending time on this right now – let's just leave it as-is and see how it behaves in real life, we can improve this workflow in future PRs.

File: .github/workflows/tx-pull.yml (hunk 1 above)
Comment (lidel) [reply .github/workflows/tx-pull.yml:24]: ack, let's leave it as-is, we can simplify in future PR.

**Review by lidel — APPROVED** [review]

**Pull request merged**